lapis/src/main/kotlin/org/genspectrum/lapis/model/AdvancedQueryCustomListener.kt

```diff
@@ -0,0 +1,534 @@
+            }
+
+            SequenceFilterFieldType.Float -> {
+                expressionStack.addLast(FloatEquals(field.name, null))
+            }
+
+            is SequenceFilterFieldType.FloatFrom -> {
+                val fieldName = field.name.split(".").first()
+                throw BadRequestException("Filter IsNull($fieldName) instead of IsNull($metadataName)", null)
+            }
+
+            is SequenceFilterFieldType.FloatTo -> {
+                val fieldName = field.name.split(".").first()
+                throw BadRequestException("Filter IsNull($fieldName) instead of IsNull($metadataName)", null)
+            }
+
+            SequenceFilterFieldType.Int -> {
+                expressionStack.addLast(IntEquals(field.name, null))
+            }
+
+            is SequenceFilterFieldType.IntFrom -> {
+                val fieldName = field.name.split(".").first()
+                throw BadRequestException("Filter IsNull($fieldName) instead of IsNull($metadataName)", null)
+            }
+
+            is SequenceFilterFieldType.IntTo -> {
+                val fieldName = field.name.split(".").first()
+                throw BadRequestException("Filter IsNull($fieldName) instead of IsNull($metadataName)", null)
+            }
+
+            is SequenceFilterFieldType.StringSearch -> {
+                val fieldName = field.name.split(".").first()
+                throw BadRequestException("Filter IsNull($fieldName) instead of IsNull($metadataName)", null)
+            }
+
+            SequenceFilterFieldType.VariantQuery -> {
+                throw BadRequestException("$VARIANT_QUERY_FIELD cannot be called from advanced query", null)
+            }
+
+            SequenceFilterFieldType.AdvancedQuery -> {
+                throw BadRequestException("$ADVANCED_QUERY_FIELD cannot be called recursively", null)
+            }
+        }
+    }
+
+    override fun enterNucleotideMutationQuery(ctx: NucleotideMutationQueryContext?) {
+        if (ctx == null) {
+            return
+        }
+        val position = ctx.position().text.toInt()
+
+        val expression = when (val secondSymbol = ctx.nucleotideMutationQuerySecondSymbol()) {
+            null -> HasNucleotideMutation(null, position)
+            else -> NucleotideSymbolEquals(null, position, secondSymbol.text.uppercase())
+        }
+
+        expressionStack.addLast(expression)
+    }
+
+    override fun exitAnd(ctx: AndContext?) {
+        val lastChildren = when (val last = expressionStack.removeLast()) {
+            is And -> last.children
+            else -> listOf(last)
+        }
+
+        val secondLastChildren = when (val secondLast = expressionStack.removeLast()) {
+            is And -> secondLast.children
+            else -> listOf(secondLast)
+        }
+
+        expressionStack.addLast(And(lastChildren + secondLastChildren))
+    }
+
+    override fun exitVariantAnd(ctx: AdvancedQueryParser.VariantAndContext?) {
```

Comment:
Why do we need `exitAnd` and `exitVariantAnd`?

Reply:
This is required because I split up variantExpr from metadataQueryExpr (this was needed to ensure MAYBE() cannot be used on metadata as it doesn't give logical responses - because of this I also need to add override exit contexts on the variantExpr options

Reply:
Why doesn't give `maybe` logical responses on metadata queries? I thought it's designed to be used on any other query. AFAIK it doesn't _do_ anything on metadata queries, but it can still be used.

Reply:
I talked with @Taepper (I added this to the PR description) - maybe does not work properly on metadata fields. It will not error but it is the same as asking if sth is true - it does not also include null fields for example - IMO this is confusing and we shouldn't let users use MAYBE on metadata fields as Maybe has a different behavior on variant queries

Reply:
I also thought that this was discussed in the last meeting adding @chaoran-chen

Reply:
https://github.com/GenSpectrum/LAPIS-SILO/blob/main/documentation/maybe_documentation.md

According to those docs, Maybe does something on mutation queries. I just checked the code: It's a no-op on all other filters (`country = Germany` is equivalent to `maybe(country = Germany)`). SILO allows to Maybe to be used wrapping any kinds of filters. I don't see a reason why we should disallow that in the advanced queries.

Is it maybe confusing because it's not properly documented?

Reply:
It doesn't make sense to use MAYBE on metadata, there you should use the exact match - so why should we allow it? 

The direct translation of MAYBE into a metadata search equivalent would be to include null fields into the result as they could "maybe" also equal to search query - this is not done. "there exists at least one possible non-ambiguous sequence symbols at the sequence position" would be "there exists at least one possible non-ambiguous field=value for that field".

I would vote very strongly to remove an option that has an unclear behavior.

Reply:
Why you might want to use it is weird nested queries: 

```
maybe( 
        123T 
     AND
        (
            789G AND country  = Germany
        ) 
    )
  OR
456C
```

Reply:
as discussed we will keep as is with more documentation- which I have now added